lms/djangoapps/teams/tests/test_csv.py

```diff
@@ -206,3 +207,130 @@ def test_team_removals_are_scoped_correctly(self):
+        cls.user_in_program = UserFactory.create()
+        cls.user_in_program_no_external_id = UserFactory.create()
+        cls.user_in_program_not_enrolled_through_program = UserFactory.create()
+
+        # user_no_program is only enrolled in the course
+        cls.add_user_to_course_program_team(cls.user_no_program, enroll_in_program=False)
+
+        # user_in_program is enrolled in the course and the program, with an external_id
+        cls.external_user_key = 'externalProgramUserId-123'
+        cls.add_user_to_course_program_team(cls.user_in_program, external_user_key=cls.external_user_key)
+
+        # user_in_program is enrolled in the course and the program, with no external_id
+        cls.add_user_to_course_program_team(cls.user_in_program_no_external_id)
+
+        # user_in_program_not_enrolled_through_program is enrolled in a program and the course, but they not connected
+        cls.add_user_to_course_program_team(
+            cls.user_in_program_not_enrolled_through_program, connect_enrollments=False
+        )
+
+        # initialize import manager
+        cls.import_manager = csv.TeamMembershipImportManager(cls.course)
+        cls.import_manager.teamset_ids = {ts.teamset_id for ts in cls.course.teamsets}
+
+    @classmethod
+    def add_user_to_course_program_team(
+        cls, user, add_to_team=True, enroll_in_program=True, connect_enrollments=True, external_user_key=None
+    ):
+        """
+        Set up a test user by enrolling them in self.course, and then optionaly:
+            - enroll them in a program
+            - link their program and course enrollments
+            - give their program enrollment an external_user_key
+        """
+        course_enrollment = CourseEnrollmentFactory.create(user=user, course_id=cls.course.id)
+        if add_to_team:
+            cls.team.add_user(user)
+        if enroll_in_program:
+            program_enrollment = ProgramEnrollmentFactory.create(user=user, external_user_key=external_user_key)
+            if connect_enrollments:
+                ProgramCourseEnrollmentFactory.create(
+                    program_enrollment=program_enrollment, course_enrollment=course_enrollment
+                )
+
+    def assert_user_on_team(self, user):
+        self.assertTrue(CourseTeamMembership.is_user_on_team(user, self.team))
+
+    def assert_user_not_on_team(self, user):
+        self.assertFalse(CourseTeamMembership.is_user_on_team(user, self.team))
+
+    def test_add_user_to_team_with_external_key(self):
+        # Make a new user with an external_user_key who is enrolled in the course and program, with an external_key,
+        #  but is not on the team
+        new_user = UserFactory.create()
+        new_ext_key = "another-external-user-id-FKQP12345"
+        self.add_user_to_course_program_team(new_user, add_to_team=False, external_user_key=new_ext_key)
+        self.assert_user_not_on_team(new_user)
+
+        with BytesIO() as mock_csv_file:
```

Comment:
Have you though of using `SimpleUploadFile` - see examples in teams/tests/test_views.py::2689

Reply:
i kinda like it this way? i like that I can just write to it and don't have to worry about concatenating strings or writing newlines or anything